reana_job_controller/htcondorvc3_job_manager.py

```diff
@@ -0,0 +1,172 @@
+# under the terms of the MIT License; see LICENSE file for more details.
+
+"""HTCondor VC3 Job Manager."""
+
+import logging
+import traceback
+import uuid
+import htcondor
+import classad
+import os
+import re
+import shutil
+import filecmp
+import pwd
+
+from retrying import retry
+from reana_job_controller.variables import (MAX_JOB_RESTARTS, SHARED_VOLUME_PATH_ROOT)
+
+from kubernetes.client.rest import ApiException
+from reana_commons.config import K8S_DEFAULT_NAMESPACE
+from reana_db.database import Session
+from reana_db.models import Workflow
+
+from reana_job_controller.job_manager import JobManager
+
+
+@retry(stop_max_attempt_number=MAX_JOB_RESTARTS)
+def submit(schedd, sub):
+    """Submit condor job to local schedd.
+
+    :param schedd: The local
+    """
+    try:
+        with schedd.transaction() as txn:
+            clusterid = sub.queue(txn)
+    except Exception as e:
+        logging.debug("Error submission: {0}".format(e))
+        raise e
+
+    return clusterid
+
+def get_input_files(workflow_workspace):
+    """Get files from workflow space.
+
+    :param workflow_workspace: Workflow directory
+    """
+    # First, get list of input files
+    input_files = []
+    for root, dirs, files in os.walk(workflow_workspace):
+        for filename in files:
+           input_files.append(os.path.join(root, filename))
+    
+    return ",".join(input_files)
+
+def get_schedd():
+    """Find and return the HTCondor sched.
+
+    :returns: htcondor schedd object.
+    """
+    schedd_ad = classad.ClassAd()
+    schedd_ad["MyAddress"] = os.environ.get("REANA_JOB_CONTROLLER_VC3_HTCONDOR_ADDR", None) 
+    schedd = htcondor.Schedd(schedd_ad)
+    return schedd
+
+def get_wrapper(workflow_workspace):
+    """Get bash job wrapper for executing on remote HPC worker node. Transfer if it does not exist.
+
+    :param workflow_workspace: Shared FS directory, e.g.: /var/reana.
+    :type workflow_workspace: str
+    """
+    wrapper = os.path.join(workflow_workspace, 'wrapper', 'job_wrapper.sh')
+    local_wrapper = '/code/files/job_wrapper.sh'
+    if os.path.exists(wrapper) and filecmp.cmp(local_wrapper, wrapper):
+        return wrapper
+    try:
+        if not os.path.isdir(os.path.dirname(wrapper)):
+            os.mkdir(os.path.dirname(wrapper))
+        shutil.copy('/code/files/job_wrapper.sh', wrapper)
+    except Exception as e:
+        logging.debug("Error transfering wrapper : {0}.".format(e))
+        logging.debug("user: {0}".format(pwd.getpwuid(os.getuid()).pw_name))
+        raise e
+    
+    return wrapper
+
+class HTCondorJobManagerVC3(JobManager):
+    """HTCondor VC3 job management."""
+
+    def __init__(self, docker_img=None, cmd=None, prettified_cmd=None,
+                 env_vars=None, workflow_uuid=None, workflow_workspace=None,
+                 cvmfs_mounts='false', shared_file_system=False,
+                 job_name=None, kerberos=False, kubernetes_uid=None):
```

Comment:
We have added in the meantime support for running unpacked Singularity images from CVMFS.  (See `unpacked_img` in `master` branch.)  Is this something that would interest you for VC3 integration as well?  If yes, this will be something to look into.  If not, we can either `raise NotImplemented` or `pass` while merging.  Asking just for the future in order to know how to propagate `v0.6.0 -> master` changes to the `HTCondorJobManagerVC3` class the best.

Reply:
In the current moment we haven't explored CVMFS very much as we are targeting mainly National Science Foundation HPC sites here in the US which will not support it, so for now when merging a `raise NotImplemented` may be best.